client/tx/factory.go

```diff
@@ -57,15 +57,35 @@ func NewFactoryCLI(clientCtx client.Context, flagSet *pflag.FlagSet) Factory {
 		signMode = signing.SignMode_SIGN_MODE_EIP_191
 	}
 
+	var gasSetting client.GasSetting
+	gasStr, err := flagSet.GetString(flags.FlagGas)
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "gas setting is not specified! using default value from client config: %s\n", clientCtx.GasSetting.String())
+		gasSetting = clientCtx.GasSetting
+	} else {
+		gasSetting, _ = client.ParseGasSetting(gasStr)
```

Comment:
Shouldn't we be checking for the error here?

Reply:
this error is not handled even before this PR: https://github.com/cosmos/cosmos-sdk/blob/2bec9d2021918650d3938c3ab242f84289daef80/client/tx/factory.go#L67

but yeah i think we should panic here.

Reply:
now i have the function panic on error: https://github.com/cosmos/cosmos-sdk/pull/12723/commits/5566bb320dfefa558a69d683b1ef0138dfc17b60